### PR TITLE
Java: Update Gradle & JDK

### PR DESCRIPTION
**Pull request by drinkbird**
* Update Gradle to 7.5.1 (+remove unnecessary local publishing target)
* Update source compatibility to JDK 11
* Refactor usage of a deprecated constructor for `Float` and fix indentation
* Update documentation for Gradle & JDK dependencies

### CODE REVIEW

File: java/src/main/java/com/xgen/interview/ShoppingCart.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public void printReceipt() {
 
         for (int i = 0; i < Array.getLength(keys) ; i++) {
             Integer price = pricer.getPrice((String)keys[i]) * contents.get(keys[i]);
-            Float priceFloat = new Float(new Float(price) / 100);
+            Float priceFloat = Float.valueOf(price) / 100;
```
Comment (jamesbroadhead) [inline java/src/main/java/com/xgen/interview/ShoppingCart.java]: this is intentionally ugly, hoping that candidates will clean it up :)

**Review by jamesbroadhead — APPROVED** [review]
`java test` on my machine works with the listed version of gradle. Thanks for the upgrade! 

1 thing to revert - otherwise looks good 

So lgtm - but from my very limited experience with gradle

File: java/src/main/java/com/xgen/interview/ShoppingCart.java (hunk 1 above)
Comment (drinkbird) [inline java/src/main/java/com/xgen/interview/ShoppingCart.java]:
```suggestion
            Float priceFloat = new Float(new Float(price) / 100);
```

Comment (drinkbird) [reply java/src/main/java/com/xgen/interview/ShoppingCart.java]: Whops, reverted

**Review by jamesbroadhead — APPROVED** [review]

**Pull request merged**